### PR TITLE
Be more permissive and helpful in the relationship check.

### PR DESCRIPTION
**Pull request by Crell**

### CODE REVIEW

**Comment by NickAnderegg** [comment]
This all likes fine to me, but it made me realize we never fixed the implementation of `__getitem__()`. If an object is missing a key, then it should raise a `KeyError` when `__getitem__` is called. In contrast, we just have it returning `None` if the key is missing. The code will definitely function, but this is slightly non-standard behavior.

Realistically, how we want to implement it is have `__getitem__` raise a `KeyError`, and then instead of a series of `if` statements, it should be a series of `try..except` blocks. Other than that implementation detail that's not directly related to this PR, the logic of this seems fine to me.

File: platformshconfig/config.py [hunk 1]
```diff
@@ -321,9 +326,9 @@ def application(self):
 
         """
 
-        if not self.is_valid_platform():
+        if not self._applicationDef():
```
Comment (NickAnderegg) [inline platformshconfig/config.py]: `_applicationDef` is a `dict` not a method

**Review by NickAnderegg — CHANGES_REQUESTED** [review]
I like this structure overall, with property-specific exceptions and just failing if the variables are absent in general, rather than a pre-requisite check. There's just one bug here, which our tests caught

File: platformshconfig/config.py (hunk 1 above)
Comment (Crell) [reply platformshconfig/config.py]: Hooray for tests!  Fixed typo.

**Review by NickAnderegg — APPROVED** [review]

**Pull request merged**